src/components/ProgressIndicator/ProgressIndicator.tsx

```diff
@@ -0,0 +1,137 @@
+  /**
+   * Optional labels for each step in the progress indicator.
+   */
+  labels?: string[];
+  /**
+   * Optional class name for additional styling.
+   */
+  className?: string;
+  /**
+   * Aria label for the progress indicator.
+   */ ariaLabelProgress?: string;
+  /**
+   * Aria label for the progress steps.
+   */
+  ariaLabelSteps?: string;
+  /**
+   * Aria labels for each step in different states.
+   */
+  ariaLabelStepCompleted?: (step: number) => string;
+  ariaLabelStepCurrent?: (step: number) => string;
+  ariaLabelStepNotStarted?: (step: number) => string;
+  /**
+   * Aria label for the completed icon.
+   */
+  ariaLabelCompletedIcon?: string;
+}
+/**
+ * ## Overview
+ *
+ * This component provides a visual representation of a multi-step process.
+ *
+ *
+ * [Figma Link](https://www.figma.com/design/kSxOhnqIhilZ9hIJd3bPgP/RW-Registration?node-id=2996-50543&m=dev)
+ * [Figma Link 2](https://www.figma.com/design/kSxOhnqIhilZ9hIJd3bPgP/RW-Registration?node-id=2529-23889&m=dev)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-progressindicator--overview)
+ */
+const ProgressIndicator = forwardRef<HTMLDivElement, ProgressIndicatorProps>(
+  (
+    {
+      steps,
+      current,
+      className,
+      ariaLabelProgress = 'Progress',
+      ariaLabelSteps = 'Progress steps',
+      ariaLabelStepCompleted = (step) => `Step ${step} completed`,
+      ariaLabelStepCurrent = (step) => `Step ${step} current`,
+      ariaLabelStepNotStarted = (step) => `Step ${step} not started`,
+      ariaLabelCompletedIcon = 'Completed Icon',
+      ...props
+    },
+    ref,
+  ) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'ProgressIndicator');
+    if (steps < 1) return null;
+
+    const getValue = (value: number) => {
+      if (value <= 0) return 0;
+      if (value >= steps) return steps;
+      return value;
+    };
+
+    return (
+      <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
+        <Progress.Root value={getValue(current)} max={steps} aria-label={ariaLabelProgress}>
+          <div className={`${baseClassName}__steps`} aria-label={ariaLabelSteps}>
+            {Array.from({ length: steps }).map((_, index) => {
+              const isComplete = current > index + 1;
+              const isCurrent = current === index + 1;
+              const stepAriaLabel = isComplete
+                ? ariaLabelStepCompleted(index + 1)
+                : isCurrent
+                  ? ariaLabelStepCurrent(index + 1)
+                  : ariaLabelStepNotStarted(index + 1);
+
+              return (
+                <Fragment key={props.labels ? props.labels[index] : index}>
+                  <div
+                    className={`${baseClassName}__item`}
+                    aria-current={isCurrent ? 'step' : undefined}
+                    aria-label={stepAriaLabel}
+                    data-testid="progress-step"
```

Comment:
if they were trying to use testid wouldn't we need this to be unique for each step?